src/plugins/jma/datasource.py

```
@@ -0,0 +1,396 @@
+            current_date = start_date or datetime.now()
+            end = end_date or datetime.now()
+            
+            # JMA AMeDAS only provides recent data (last 48-72 hours)
+            max_history = datetime.now() - timedelta(days=3)
+            
+            # Convert to naive datetime for comparison if needed
+            if current_date.tzinfo is not None:
+                current_date = current_date.replace(tzinfo=None)
+            if end.tzinfo is not None:
+                end = end.replace(tzinfo=None)
+                
```

Pylint (3.3.7)

[convention] Trailing whitespace

(C0303)